File: guess game.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{

    int a,b,c,d;

    // this is game is about guessing the number

    printf("\t \t \t \t-----------------------GUESS THE NUMBER------------------------\n");
    printf("\t \t \t \t-----------------------GUESS THE NUMBER------------------------\n");
    printf("\t \t \t \t-----------------------GUESS THE NUMBER------------------------\n");
    printf("\t \t \t \t-----------------------GUESS THE NUMBER------------------------\n \n \n");

    //choosing difficulty
    diffi:
    printf("\n \n \t \t \t \t-----------------------CHOOSE DIFFICULTY------------------------\n");
    printf("\t \t \t \t \t \t0 --NOOB-------------\n");
    printf("\t \t \t \t \t \t1 --EASY-------------\n");
    printf("\t \t \t \t \t \t2 --MEDIUM-----------\n");
    printf("\t \t \t \t \t \t3 --HARD-------------\n");
    printf("\t \t \t \t \t \t4 --EXTREME----------\n \n");

    printf("\t \t \t \t \t \tPLEASE CHOSE ONE:\t");
    scanf("%d",&a);

    if(a== 0){
        printf("\t \t \t \t \t \t\n \nYOU HAVE CHOSEN NOOB\n \n");
    }
    else if(a== 1){
        printf("\t \t \t \t \t \t\n \nYOU HAVE CHOSEN EASY\n \n");
    }
    else if(a == 2){
        printf("\t \t \t \t \t \t\n \nYOU HAVE CHOSEN MEDIUM\n");
    }
    else if(a == 3){
        printf("\t \t \t \t \t \t\n \nYOU HAVE CHOSEN HARD\n");
    }
    else if(a == 4){
        printf("\t \t \t \t \t \t\n \nYOU HAVE CHOSEN EXTREME\n");
    }
    else{
        printf("\t \t \t \t \t \t\n \nwrong input, try again\n");
        goto diffi;
    };

    //condition for the game starts here
    here:
     if(a==0){
        printf("\t \t \t \t \t \tGuess the number\n");
        printf("\t \t \t \t \t \tyou can only chose from 0 to 5\n");
        printf("\t \t \t \t \t \twhat is the number:\t");
        scanf("%d",&b);
    }
    else if(a==1){
        printf("\t \t \t \t \t \tGuess the number\n");
        printf("\t \t \t \t \t \tyou can only chose from 0 to 10\n");
        printf("\t \t \t \t \t \twhat is the number:\t");
        scanf("%d",&b);
    }

    else if(a==2){
        printf("\t \t \t \t \t \tGuess the number\n");
        printf("\t \t \t \t \t \tyou can only chose from 0 to 100\n");
        printf("\t \t \t \t \t \twhat is the number:\t");
        scanf("%d",&b);
    }
    else if(a==3){
        printf("\t \t \t \t \t \tGuess the number\n");
        printf("\t \t \t \t \t \tyou can only chose from 0 to 1000\n");
        printf("\t \t \t \t \t \twhat is the number:\t");
        scanf("%d",&b);
    }
    else if(a==4){
        printf("\t \t \t \t \t \tGuess the number\n");
        printf("\t \t \t \t \t \tyou can only chose from 0 to 10000\n");
        printf("\t \t \t \t \t \twhat is the number:\t");
        scanf("%d",&b);
    }
// for noob
if (a==0){
    srand((unsigned)time(NULL));

    for(int x = 1; x < 10; x++){
        int random = rand() % 6;
        printf("%d\n", random);

        if( b == random ){
            printf("\t \t \t \t \t \t\nyou guessed it correctly\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            iaa:
                if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tChange difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto iaa;
                }
        }
        else{
            printf("\t \t \t \t \t \t\nyou did not get the value\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            i0:
            if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto i0;
                }
        }
    }
 }

 // for easy
 if (a==1){
    srand((unsigned)time(NULL));

    for(int x = 1; x < 10; x++){
        int random = rand() % 11;
        printf("%d", random);

        if( b == random ){
            printf("\t \t \t \t \t \t\nyou guessed it correctly\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            ia:
                if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tChange difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto ia;
                }
        }
        else{
            printf("\t \t \t \t \t \t\nyou did not get the value\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            i1:
            if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto i1;
                }
        }
    }
 }

// for medium
     if (a==2){
    srand((unsigned)time(NULL));

    for(int x = 1; x < 10; x++){
        int random = rand() % 101;
        printf("%d", random);

        if( b == random ){
            printf("\t \t \t \t \t \t\nyou guessed it correctly\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            ib:
                if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto ib;
                }
        }
        else{
            printf("\t \t \t \t \t \t\nyou did not get the value\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            i2:
            if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto i2;
                }
        }
    }
 }


// for hard
     if (a==3){
    srand((unsigned)time(NULL));

    for(int x = 1; x < 10; x++){
        int random = rand() % 1001;
        printf("%d", random);

        if( b == random ){
            printf("\t \t \t \t \t \t\nyou guessed it correctly\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            ic:
                if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto ic;
                }
        }
        else{
            printf("\t \t \t \t \t \t\nyou did not get the value\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            i3:
            if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto i3;
                }
        }
    }
 }

// for extreme
     if (a==4){
    srand((unsigned)time(NULL));

    for(int x = 1; x < 10; x++){
        int random = rand() % 10001;
        printf("%d", random);

        if( b == random ){
            printf("\t \t \t \t \t \t\nyou guessed it correctly\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            id:
                if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto id;
                }
        }
        else{
            printf("\t \t \t \t \t \t\nyou did not get the value\n");
            printf("\t \t \t \t \t \tDo you want to play again?\n");
            printf("\t \t \t \t \t \t1 for yes\n");
            printf("\t \t \t \t \t \t2 for no\n");
            printf("\t \t \t \t \t \t3 to change difficulty\n");
            printf("\t \t \t \t \t \tinput value:\t");
            scanf("%d",&d);

            i4:
            if(d==1){
                    printf("\t \t \t \t \t \t\nok lets go\n");
                    goto here;
                }
                else if(d==2){
                    printf("\t \t \t \t \t \tthanks for playing\n");
                    return 0;
                }
                else if(d==3){
                    printf("\t \t \t \t \t \tto change difficulty\n");
                    goto diffi;
                }
                else{
                    printf("\t \t \t \t \t \twrong input\n");
                    goto i4;
                }
        }
    }
 }

    return 0;
}

File: Guess_Game.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Main.java to edit this template
 */
package guess_game;

//gui would be added soon

import java.util.Scanner; //takes input with scanner..
import java.util.Random; // gives random numbers..

/**
 *
 * @author ohanu
 */
public class Guess_Game {
    public static void main(String[] args) {
    
        //int play;
     
        System.out.println("Hi, Welcome to my Guess Game\n");
        System.out.println("----------------GUESS THE NUMBER------------");
        System.out.println("----------------GUESS THE NUMBER------------");
        int play = 1;
        //play = 0;
        while(play == 1){
        System.out.println("----------------Choose Difficulty------------");
        System.out.println("--1 ------Easy---------");
        System.out.println("--2 ------Medium-------");
        System.out.println("--3 ------Hard---------");
        System.out.println("--4 ------Extreme------");
        System.out.println("input: "); 
        Scanner scanner = new Scanner(System.in);
            //input for difficulties 
                int guess = scanner.nextInt();
                if(guess == 1){
                    System.out.println("easy");
                    System.out.println("Guess the number 0 - 9");
                    System.out.println("input: "); 
                    int gue = scanner.nextInt();
                    System.out.println("you choose: "+gue);
                    //for cpu
                    Random random = new Random();
                    int randno = random.nextInt(8) + 1;
                    System.out.println("CPU choose: "+randno);
                        if(gue == randno){
                            System.out.println("correct");
                            System.out.println("want to play again?");
                            System.out.println("input: "); 
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no"); 
                            play = scanner.nextInt();
                            }
                        else{
                            System.out.println("incorrect");
                            System.out.println("want to play again?");
                            System.out.println("input: "); 
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no");
                            play = scanner.nextInt();
                        }
                        if( play == 1 ){
                            System.out.println("lets continue");
                                    }
                         else{
                            System.out.println("thanks for playing");
                            break;
                                }
                }
                else if(guess == 2){
                    System.out.println("medium");
                    System.out.println("Guess the number 0 - 19");
                    System.out.println("input: "); 
                    int gue = scanner.nextInt();
                    System.out.println("you choose: "+gue);
                    //for cpu
                    Random random = new Random();
                    int randno = random.nextInt(18) + 1;
                    System.out.println("CPU choose: "+randno);
                        if(gue == randno){
                            System.out.println("correct");
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no");
                            System.out.println("input: "); 
                            play = scanner.nextInt();
                            }
                        else{
                            System.out.println("incorrect");
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no");
                            System.out.println("input: "); 
                            play = scanner.nextInt();
                        }
                        if( play == 1 ){
                            System.out.println("lets continue");
                                    }
                         else{
                            System.out.println("thanks for playing");
                            break;
                                }
                }
                else if(guess == 3){
                    System.out.println("Hard");
                    System.out.println("Guess the number 0 - 29");
                    System.out.println("input: "); 
                    int gue = scanner.nextInt();
                    System.out.println("you choose: "+gue);
                    //for cpu
                    Random random = new Random();
                    int randno = random.nextInt(28) + 1;
                    System.out.println("CPU choose: "+randno);
                        if(gue == randno){
                            System.out.println("correct");
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no");
                            System.out.println("input: "); 
                            play = scanner.nextInt();
                            }
                        else{
                            System.out.println("incorrect");
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no");
                            System.out.println("input: "); 
                            play = scanner.nextInt();
                        }
                        if( play == 1 ){
                            System.out.println("lets continue");
                                    }
                         else{
                            System.out.println("thanks for playing");
                            break;
                                }
                }
                else if(guess == 4){
                    System.out.println("Extreme");
                    System.out.println("Guess the number 0 - 39");
                    System.out.println("input: "); 
                    int gue = scanner.nextInt();
                    System.out.println("you choose: "+gue);
                    //for cpu
                    Random random = new Random();
                    int randno = random.nextInt(38) + 1;
                    System.out.println("CPU choose: "+randno);
                        if(gue == randno){
                            System.out.println("correct");
                            System.out.println("want to play again");
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no");
                            System.out.println("input: "); 
                            play = scanner.nextInt();
                            }
                        else{
                            System.out.println("incorrect");
                            System.out.println("1 -- yes");
                            System.out.println("2 -- no");
                            System.out.println("input: "); 
                            play = scanner.nextInt();
                        }
                        if( play == 1 ){
                            System.out.println("lets continue");
                                    }
                         else{
                            System.out.println("thanks for playing");
                            break;
                                }
                }
                else{
                    System.out.println("Wrong input\nDo you wat to try again?");
                    System.out.println("1 -- yes");
                    System.out.println("2 -- no");
                    System.out.println("input: "); 
                    play = scanner.nextInt();
                }
                if( play == 1 ){
                            System.out.println("lets continue");
                                    }
                         else{
                            System.out.println("thanks for playing");
                            break;
                                }
                
               
        }      
            
    }
    
}
